bcgov/biohubbc
# BHBC-2166: Enhancements to EML Function to Support Project/Survey EML Generation

# Overview
The current EML code builds an EML json object for a project and all of its surveys. We need to modify this code so that we have 2 distinct functions, one that builds an EML object for a project and one that builds an EML object for a survey.

## Links to Jira tickets

 - https://quartech.atlassian.net/browse/BHBC-2166

## Description of relevant changes

 - `eml-service` is enhanced to support a function for building EML for a single project, which includes references to any surveys under the related projects section.

 - `eml-service` is enhanced to support a function for building EML for a single survey, which includes a reference to the parent project under the related projects section.

 - All “system” node attributes are to be left as empty strings, as this will be populated by BioHub. Example:

```xml

<dataset system="https://sims.nrs.gov.bc.ca" id="a3659666-0879-4b4f-aedb-467891db198a">


<dataset system="" id="a3659666-0879-4b4f-aedb-467891db198a">
```

 - All “TaxonID” node “provider” attributes are to be left as empty strings, and the value of the node now reflects the primary key of taxonomic code. Example:

```xml

<taxonId provider="https://sims.nrs.gov.bc.ca">M-ALAM</taxonId>


<taxonId provider="">2065</taxonId>
```

 - Additionally, removed the following nodes:
   - `eml.access` node
   - `eml.dataset.metadataProvider`
   - `additionalMetadata` nodes associated with `proprietaryDataCategory`
   - `additionalMetadata` nodes associated with `proprietorName`
   - `additionalMetadata` nodes associated with `proprietaryDataCategoryRationale`
   - `additionalMetadata` nodes associated with `dataSharingAgreementRequired`

## How to Test
1. Create a project and a survey.
2. Visit a survey.
3. Click Submit Data
4. Check the API logs. The project EML and survey EML strings should be printed
